dev/tasks/r/github.packages.yml

```diff
@@ -288,11 +281,18 @@ jobs:
           if [[ "$DEVTOOLSET_VERSION" -gt 0 ]]; then
             # enable the devtoolset version to use it
             source /opt/rh/devtoolset-$DEVTOOLSET_VERSION/enable
+            mkdir -p ~/.R
+            echo "CXX17=g++ -std=g++17 -g -O2 -fpic" >> ~/.R/Makevars

```

Comment:
Should this be added to `ci/scripts/r_docker_configure.sh`? (Used here in L271 above)

Reply:
I thought about that but as this is only an issue with this specific build (afaik we don't run centos7 builds anywhere else?) I decided against it to keep it local&visible to the problem. Happy to move it though.

Reply:
There's already centos-specific code in the script to handle the devtoolset though, seems like this would necessarily go with it, right? https://github.com/apache/arrow/blob/master/ci/scripts/r_docker_configure.sh#L72

Reply:
moved it